adagrams/game.py

```diff
@@ -1,11 +1,80 @@
+import random
+
 def draw_letters():
-    pass
+    
+    letters   = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3, 'H': 2, 
+    'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6, 'O': 8, 'P': 2, 
+    'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1, 
+    'Y': 2, 'Z': 1 }
+
+    pieces = []
+    # number of pieces in letter pool
+    count = 98
+
+    for i in range(10):
+        rand_piece = random.randrange(1, count+1)
+        key = weighted(letters, rand_piece)
+        pieces.append(key)
+        
+        letters[key] -=1
+        if letters[key] == 0:
+            letters.pop(key)
+        count -= 1
+    return pieces
 
+def weighted(dict,random_piece):
+    for char, weight in dict.items():
+            random_piece -= weight
+            if random_piece <= 0:
+                return char
+            
 def uses_available_letters(word, letter_bank):
-    pass
+    # check if they can use the word they inputed
+    new_letter_bank = []
+    for l in letter_bank:
+        new_letter_bank.append(l.upper())
+    for letter in word:
+        if letter.upper() in new_letter_bank:
+            new_letter_bank.remove(letter.upper())
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    values = {('A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T') : 1,
+            ('D', 'G'):	2,
+            ('B', 'C', 'M', 'P'): 3,
+            ('F', 'H', 'V', 'W', 'Y') : 4,
+            ('K'): 5,
+            ('J', 'X'):	8,  
+            ('Q', 'Z'):	10}
+    
+    sum = 0
+    if len(word) >= 7:
+        sum += 8
+    for letter in word:
+        for val in values.keys():
+            if letter.upper() in val:
+                sum += values.get(val)
```

Comment:
Nothing wrong with how you accessed the `value` on the `key/value` pair for the dictionary, I just want to show you that in industry you will probably see it done like `values[val]`. Also you might want to change your temp variable to something a bit more accurate like `key` since you are looping through the keys of `values`.